src/utils/daos_hdlr.c

```diff
@@ -2464,8 +2467,35 @@ dfuse_count_query(struct cmd_args_s *ap)
 	ap->dfuse_mem.container_count = query.container_count;
 	ap->dfuse_mem.found           = query.found;
 
+	D_ALLOC_ARRAY(stat, query.stat_count);
+	if (stat == NULL)
+		goto close;
```

Comment:
Do we need this?
```
rc = daos_errno2der(errno);
```

Reply:
Actually no, if this fails then the ioctl will still work but only include the other metrics, not the stats.  I can document or revise this behaviour later if you prefer though.

Reply:
If we fail to allocate something, we're out of memory, right? So isn't that sort of "fatal"?

Reply:
The command will fetch two things, the inode counts which it already does and the per-container statistics which this PR adds, if this allocation fails then the inode counts will still be reported but the per-container stats won't be.

This code would therefore work as-is, but you're right, it would probably be better to return an error here rather than half the data.

Reply:
Fixed.